hackathon_site/dashboard/frontend/src/components/general/SideSheetRight/SideSheetRight.test.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import SideSheetRight from "./SideSheetRight";
+import {
+    render,
+    fireEvent,
+    waitForElementToBeRemoved,
+    waitFor,
+    waitForElement,
+} from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+describe("SideSheetRight", () => {
+    const testDetail = {
+        name: "Arduino",
+        type: "red",
+        total: 30,
+        available: 29,
+        img: "https://www.filterforge.com/more/help/images/size200.jpg",
+        category: ["MCU", "FPGA"],
+        manufacturer: "Canakit",
+        model_num: "Model 3B+",
+        datasheet: "link",
+        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
+        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
+        quantity: 3,
+    };
+
+    const addCartMock = jest.fn();
+
+    test("render", () => {
+        const { asFragment } = render(
+            <SideSheetRight detail={testDetail} addCartFunction={addCartMock} />
+        );
+
+        expect(asFragment()).toMatchSnapshot();
```

Comment:
Would also get rid of the snapshot test, it's not super helpful. Some good things to test would be:
- Displays the children
- Displays the title

And all the other stuff about visibility in my other comments.